File: src/Yumney.Recipes.Application/Commands/SaveRecipeRequest.cs
namespace Yumney.Recipes.Application.Commands;

public sealed record SaveRecipeRequest(
    string Title,
    string? Description,
    List<SaveRecipeIngredientRequest> Ingredients,
    List<SaveRecipeStepRequest> Steps,
    int? Servings,
    int? PrepTimeMinutes,
    int? CookTimeMinutes,
    string? Difficulty,
    string? ImageUrl,
    string? SourceUrl = null);

public sealed record SaveRecipeIngredientRequest(string Name, decimal? Amount, string? Unit);

public sealed record SaveRecipeStepRequest(int Number, string Description);

File: client/libs/shared/api-recipes/src/lib/save-recipe-request.ts
export interface SaveRecipeRequest {
  title: string;
  description: string | null;
  ingredients: { name: string; amount: number | null; unit: string | null }[];
  steps: { number: number; description: string }[];
  servings: number | null;
  prepTimeMinutes: number | null;
  cookTimeMinutes: number | null;
  difficulty: string | null;
  imageUrl: string | null;
  sourceUrl?: string;
  tags?: string[];
}
